### PR TITLE
chore: nullable sender [WPB-1815]

### PR DESCRIPTION
**Pull request by Garzas**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Some federation events will not contain senderId so we need to make this variable nullable in our database

Event will look like

`{
  "type": "federation.delete",
  "domains": ["backendB.com", backendC.com"]
} `

And system message will not contain senderUserId because we need to create system message locally for each of affected domain users

### Solutions

Migrate messages to new table with nullable senderId field

### CODE REVIEW

**Comment by MohamadJaara** [comment]
this is dangers change, can you please explain a bit more about those event and that are the data provided 
maybe we can provide a work around by having self user as the sender or even a fixed is like 
UserId("server", ${serverDomain})

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/ephemeral/DeleteEphemeralMessageForSelfUserAsReceiverUseCase.kt [hunk 1]
```diff
@@ -74,7 +74,7 @@ internal class DeleteEphemeralMessageForSelfUserAsReceiverUseCaseImpl(
                         sendDeleteMessageToOriginalSender(
                             message.id,
                             message.conversationId,
-                            message.senderUserId,
+                            message.senderUserId!!, // TODO Mohamad can we somehow secure this case from getMessageById
```
Comment (MohamadJaara) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/ephemeral/DeleteEphemeralMessageForSelfUserAsReceiverUseCase.kt:77]: you can do the check before calling sendDeleteMessageToOriginalSender

```kotlin
message.senderUserId?.also {
sendDeleteMessageToOriginalSender()

}
```

**Comment by typfel** [comment]
> this is dangers change, can you please explain a bit more about those event and that are the data provided
> maybe we can provide a work around by having self user as the sender or even a fixed is like
> UserId("server", ${serverDomain})

I agree, for system messages I think it's very reasonable to assign the sender to the self user.

**Comment by Garzas** [comment]
Closing because we want to have non nullable senderUserId

**Pull request closed without merging**